Add chapter 14 Logical Vectors notes (#80)
r4ds/bookclub-r4ds: 14-logical_vectors.Rmd

# Logical Vectors

**Learning objectives:**

> -   What are Vectors and Why we talking about Logical Vectors

Reviewer: ```suggestion
-   What are Vectors and Why are we talking about Logical Vectors
```

-   How to use logical vectors & Boolean Algebra in

    -   Comparison

    -   Identifying missing values

> -   What are the logical summery functions we could leverage in R?

Reviewer: ```suggestion
-   What are the logical summary functions we could leverage in R?
```

-   Conditional Transformations using `if_else()` and `case_when()` functions

> ## What is logical Vectors

Reviewer: ```suggestion
## What are Logical Vectors
```

-   Vectors in R are the same as the arrays in C language, which are used to hold multiple data values of the same type

-   The mental model for using Logical Vectors

    -   Defining constraints on your data to answer Binary Questions

## How to use logical vectors & Boolean Algebra

![](images/14_venn_diagrams.png)

-   Combining logical vectors using Boolean algebra

>     -   Venn Diagrams here shows the logical relations between logical vectors

Reviewer: ```suggestion
    -   Venn diagram here shows the logical relations between logical vectors
```

### Comparison

-   numeric comparison creates logical vectors, and you can see that in the following example

```{r}
library(tidyverse)
library(nycflights13)
```

> -   There is an intermediate execution step that form the logical vectors behind the scene when you use `filter()`

Reviewer: ```suggestion
-   There is an intermediate execution step that forma the logical vector behind the scenes when you use `filter()`
```

```{r}
flights |> filter(dep_time > 600 & dep_time < 2000 & abs(arr_delay) < 20)
```

-   To separate this intermediate step in it's own variable we use `mutate`

```{r}
flights |> mutate( daytime = dep_time > 600 & dep_time < 2000, approx_ontime = abs(arr_delay) < 20, .keep = "used" )
```

> -   R doesn't round by default the float numeric value, but R console is rounding close number to integers

Reviewer: ```suggestion
-   R doesn't round the float numeric value by default, but R console is rounding close number to integers
```
